src/search/srde.c

```
@@ -0,0 +1,274 @@
+  cfg->crossover = (real)0.9;
+  cfg->init_span = (dpos > 0.0) ? dpos : (real)1.0;
+  cfg->seed = 0;
+}
+
+static real sr_de_rand_span(sr_rng *rng, real span)
+{
+  const real u = (real)sr_rng_uniform01(rng);
+  return (u * (real)2.0 - (real)1.0) * span;
+}
+
+static int sr_de_pick_indices(sr_rng *rng, int pop, int skip,
```

Clang (14.0.6)

[warning] 2 adjacent parameters of 'sr_de_pick_indices' of similar type ('int') are easily swapped by mistake

(bugprone-easily-swappable-parameters)